.azure-pipelines.yml

```diff
@@ -75,6 +75,28 @@ jobs:
       python --version
       pip install pre-commit
     displayName: 'Bootstrap'
+  - bash: |
+      set -eux
+      # Commit signature is not mandatory for the repo (yet ?), however we
+      # want to make sure the signed commits are actually valid.
+      # Indeed, it seems the signature process breaks from time to time,
+      # see for instance commit 1e1535b010050c025d85b685e96e08db5e9cd7bb
+      # that is considered invalid, but has been generated 3mn after the valid
+      # 2df2c661a4774097615b7f66078207b15a56316b on the same computer with
+      # the same key...)
+      python -c "
+      from urllib.request import urlopen
+      import json
+      import re
+      match = re.match(r'refs/pull/([0-9]+)/merge', '$(Build.SourceBranch)')
+      if match:
+        pr_id = match.group(1)
+        r = urlopen(f'https://api.github.com/repos/Scille/parsec-cloud/pulls/{pr_id}/commits')
+        bad_commits = [c['sha'] for c in json.load(r) if c['commit']['verification']['reason'] == 'invalid']
+        if bad_commits:
+          raise SystemExit(f'''Invalid signatures in commits: {', '.join(bad_commits)}''')
+      "
```

Comment:
Maybe move to a dedicated python script and use:
```bash
./misc/check_commit_signatures_for_pr $(System.PullRequest.PullRequestNumber)
```

Reply:
For the moment I only consider this code as a little sanity check so it's fine to inline it in the azure pipeline (we shouldn't need this as a separate script because gpg shouldn't have created this bad commit in the first place).

Reply:
I don't really agree, this CI file is already 600 lines long and hard to navigate. Also inline code is hard to read for a couple of reasons:
- no syntax highlighting and other IDE support
- it's a bash string, with all the quote and substitution mess that comes with it

I'm not going to fight over it though, I already validated this PR so it's fine if you want to merge it as it is :)

Reply:
The script is 10 lines long, almost half of the lines are for a comment explaining why this mess is needed.
I'm not convinced it would be more readable in a separate file (given you'd have to switch between the two, with the explanation comment missing in one of the places)

Of course it would be must better if we could split this file between jobs ;-)